emailpal/__init__.py

```diff
@@ -1 +1,85 @@
+import abc
+import re
+from typing import Dict, TypeVar, Generic, Dict, Any, cast  # NOQA
+from django.utils.safestring import SafeString
+from django.core.mail import EmailMultiAlternatives
+from django.template.loader import render_to_string
+from django.utils.html import strip_tags
+
 VERSION = '0.0.1'
+
+T = TypeVar('T')
+
+
+def collapse_and_strip_tags(text: str) -> str:
+    '''
+    Strips HTML tags and collapases newlines in the given string.
+
+    Example:
+
+      >>> collapse_and_strip_tags('\\n\\n<p>hi james</p>\\n\\n\\n')
+      '\\nhi james\\n'
+    '''
+
+    return re.sub(r'\n+', '\n', strip_tags(text))
+
+
+class SendableEmail(Generic[T], metaclass=abc.ABCMeta):
+    example_ctx = None  # type: T
+
+    @property
+    @abc.abstractmethod
+    def subject(self) -> str:
+        pass  # pragma: no cover
+
+    @property
+    @abc.abstractmethod
+    def template_name(self) -> str:
+        pass  # pragma: no cover
+
+    def _cast_to_dict(self, ctx: T) -> Dict[str, Any]:
+        return cast(Dict[str, Any], ctx)
```

Comment:
We should actually ensure `ctx` is a dict here and raise a helpful exception if it isn't.